src/js/model/Score.js

```diff
@@ -0,0 +1,71 @@
+function Score() {
+  this._scoreboard = [];
+  this._total = 0;
+
+  this._rolls = [];
+}
+
+Score.prototype.getScoreBoard = function() {
+  return this._scoreboard;
+};
+
+Score.prototype.getScoreTotal = function() {
+  return this._total;
+};
+
+Score.prototype.getRolls = function() {
+  return this._rolls;
+};
+
+Score.prototype.rollNumber = function() {
+  return this.getRolls().length;
+};
+
+Score.prototype.rollMinusOne = function() {
+  return this.getRolls().slice(-2)[0];
+};
+
+Score.prototype.rollMinusTwo = function() {
+  return this.getRolls().slice(-3)[0];
+}
+
+Score.prototype.addBonus = function(roll) {
+  this._scoreboard[this._scoreboard.length - 2] += roll;
+}
+
+Score.prototype.recordRolls = function(roll) {
+  this._rolls.push(roll);
+};
+
+Score.prototype.setScore = function(roll, last_frame, this_frame, game_length) {
+  this.recordRolls(roll);
+  this.scorePins(roll, this_frame);
+  this.scoreSpare(roll, this_frame, last_frame);
+  this.scoreStrike(roll, game_length);
+  this._total = this._scoreboard.reduce(function(a,b) { return a + b });
+};
+
+Score.prototype.scorePins = function(roll, current_frame) {
+  if (current_frame.isNew()) { this._scoreboard.push(roll); }
+  else { this._scoreboard[this._scoreboard.length - 1] += roll; }
+}
+
+Score.prototype.scoreSpare = function(roll, current_frame, last_frame) {
+  if (typeof last_frame !== 'undefined') {
+    if (last_frame.isSpare() && current_frame.isNew()) { this._scoreboard[this._scoreboard.length - 2] += roll;; }
```

Comment:
Super nice use of private methods and variable names to enhance readability here though. Makes it really easy to see what's going on. Though, perhaps, the `this._scoreboard` line which is a little more obscure.

I'd also look into JS variable name conventions — usually they're written as `currentFrame` rather than `current_frame`. JSHint or similar might be able to remind you of this too.